metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -493,10 +521,110 @@ def emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
+            version=0,
+            hash="",
+            platformSchema=MySqlDDL(tableSchema=""),
+            fields=self.gen_schema_fields(column_data),
+        )
+        return schema_metadata
+
+    def gen_dataset_urn(self, datahub_dataset_name: str) -> str:
+        return make_dataset_urn_with_platform_instance(
+            platform=self.platform,
+            name=datahub_dataset_name,
+            platform_instance=self.config.platform_instance,
+            env=self.config.env,
+        )
+
+    def construct_dataset_from_dataset_data(
+        self, dataset_data: dict
+    ) -> DatasetSnapshot:
+        dataset_response = self.get_dataset_info(dataset_data.get("id"))
+        dataset = SupersetDataset(**dataset_response["result"])
+        datasource_urn = self.get_datasource_urn_from_id(
+            dataset_response, self.platform
+        )
+        if dataset.changed_on_utc:
+            modified_dt = dp.parse(dataset.changed_on_utc)
+        else:
+            modified_dt = datetime.now(timezone.utc)
+
+        modified_ts = int(modified_dt.timestamp() * 1000)
+        dataset_url = f"{self.config.display_uri}{dataset.explore_url or ''}"
+
+        dataset_info = DatasetPropertiesClass(
+            name=dataset.table_name,
+            description="",
+            lastModified=TimeStamp(time=modified_ts),
+            externalUrl=dataset_url,
+        )
+        aspects_items: List[Any] = []
+        aspects_items.extend(
+            [
+                self.gen_schema_metadata(dataset_response),
+                dataset_info,
+            ]
+        )
+
+        dataset_snapshot = DatasetSnapshot(
+            urn=datasource_urn,
+            aspects=aspects_items,
+        )
+        return dataset_snapshot
+
+    def emit_dataset_mces(self) -> Iterable[MetadataWorkUnit]:
+        for dataset_data in self.paginate_entity_api_results("dataset", PAGE_SIZE):
+            try:
+                dataset_snapshot = self.construct_dataset_from_dataset_data(
+                    dataset_data
+                )
+                mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+                yield MetadataWorkUnit(id=dataset_snapshot.urn, mce=mce)
+                yield from self._get_domain_wu(
+                    title=dataset_data.get("table_name", ""),
+                    entity_urn=dataset_snapshot.urn,
+                )
+            except Exception as e:
+                logger.warning(f"Dataset full logs: {dataset_data}")
+                logger.warning(
+                    f"Failed to emit dataset MCE {dataset_data.get('table_name')}. Error: \n{e}"
+                )
+                continue
 
     def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
-        yield from self.emit_dashboard_mces()
-        yield from self.emit_chart_mces()
+        # yield from self.emit_dashboard_mces()
+        # yield from self.emit_chart_mces()
+        yield from self.emit_dataset_mces()
```

Comment:
can we add a bool config like `ingest_datasets` - and make that default false until the lineage changes land in a future PR

Reply:
Yes, I will look to add that in 👍

Reply:
Also added `ingest_dashboards`, `ingest_charts` where their default is set to true